File: src/userservice/userservice/crypto.py
import bcrypt


def generate_hash(password):
    return bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')


def validate_password(password, password_hash):
    if isinstance(password_hash, str):
        password_hash = password_hash.encode('utf-8')
    return bcrypt.checkpw(password.encode('utf-8'), password_hash)

File: src/userservice/src/crypto.rs
use bcrypt::{hash, verify, DEFAULT_COST};

pub fn generate_hash(password: &str) -> Result<String, bcrypt::BcryptError> {
    hash(password, DEFAULT_COST)
}

pub fn validate_password(password: &str, hash_str: &str) -> Result<bool, bcrypt::BcryptError> {
    verify(password, hash_str)
}
